### ISSUE TITLE
[FEATURE] add adaptive learning rate feature (automatic adjust LR-level)

### ISSUE THREAD

**Issue by yovelop** [sdatkinson/neural-amp-modeler#160]
Now with every epoch learning rate decreased according to lr_decay value by ExponentialLR rule.
When users make their nam-models they trying to achieve maximum quality of model no matter how long will it train.
So they run trainig for 1000+ epochs and after 500-600 epochs they see completely flat horisontal line of error with no next improvement no matter how many epochs there will be.
But lowering LR_decay can get it to better results.

So:

1) Automatic set lr_decay value according to lr and epochs.
For example:
lr_decay ( lr, epochs ) = minimum( lr_decay_default_value , lr_decay_default_value  /  epochs * 500  )
, it's very simple and stupid but efficient method
, now lr_decay by default = 0.007
On some drive marshall training tests i get results:
    standart architecture (lr_dec = 0.007, 2000 ep) = 0.0020 ESR
    lite architecture(lr_dec = 0.002, 2000 ep)          = 0.0017 ESR
 
2) Now NAM using ExponentialLR, maybe try other, for example:
lr_scheduler.ReduceLROnPlateau (https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html#torch.optim.lr_scheduler.ReduceLROnPlateau)
[lr_scheduler.CyclicLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CyclicLR.html#torch.optim.lr_scheduler.CyclicLR)

3) In many my neural training models i use adaptive learning rate. 
it decreasing lr when many epochs there is no significant error improvement but big amplitude of error level (reduceLROnPlateau analog)
, and increasing lr when many epochs there is no significant error improvement and very small amplitude of error level (by batches or epochs).

**Comment by 2-dor** [comment]
> 

Brilliant tip!

**Comment by sdatkinson** [comment]
I've tried a lot of these in the past :) Exponential works fairly well.

I don't think I'm keen to open this up on the "user-friendly" trainers, but if you want to get deep like this, the learning rate is configurable using the script-based trainer, `bin/train/main.py`.  You configure this under the model's configuration JSON. See [`Model.configure_optimizers()`](https://github.com/sdatkinson/neural-amp-modeler/blob/1d354a00dc75719e49624cb5af296ee66d5e5155/nam/models/base.py#L188).
